packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx

```diff
@@ -12,9 +12,16 @@ export interface Props extends ButtonProps {
 }
 
 export function ClipboardButton({ onClipboardCopy, onClipboardError, children, getText, ...buttonProps }: Props) {
+  const [buttonText, setButtonText] = useState(children);
+  const [buttonIcon, setButtonIcon] = useState(buttonProps.icon);
+
+  buttonProps.icon = buttonIcon;
+
   const buttonRef = useRef<null | HTMLButtonElement>(null);
   const copyTextCallback = useCallback(async () => {
     const textToCopy = getText();
+    setButtonText('Copied!');
+    setButtonIcon('check');
 
     try {
       await copyText(textToCopy, buttonRef);
```

Comment:
Also, we should set the success state once we actually successfully copy text - after the `await copyText` call!